docs/src/whatsnew/index.rst

```diff
@@ -10,11 +10,7 @@ Iris versions.
 .. toctree::
    :maxdepth: 1
 
-   3.1.0.rst
-   3.0.4.rst
-   3.0.3.rst
-   3.0.2.rst
-   3.0.1.rst
```

Comment:
@pp-mo  Do you know why we had separate what's news for each bug release?
Was it just there to draw attention to the fact that there multiple 3.0.x releases?

Reply:
I don't know really.
However, I did find that we did _not_ do this with the 2.2.1 bugfix release : there we just has a [modified 2.2.rst file](https://github.com/SciTools/iris/tree/v2.2.1/docs/iris/src/whatsnew)

Reply:
Also, the 3.0.4.rst incorporates all the entries from the other 3